File: Backend/Todo/Todo.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace Backend;

public class Todo : IEquatable<Todo>
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int TodoId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public bool Done { get; set; }
    
    public Todo() {}

    public Todo(int todoId, string name, string description, bool done)
    {
        TodoId = todoId;
        Name = name;
        Description = description;
        Done = done;
    }

    public bool Equals(Todo obj)
    {
        return TodoId == obj.TodoId
            && Name == obj.Name
            && Description == obj.Description
            && Done == obj.Done;
    }

    public override string ToString()
    {
        return JsonConvert.SerializeObject(this);
    }
}

File: Frontend/lib/todo/todo.dart
class Todo {
  int TodoId;
  String Name;
  String Description;
  bool Done;

  Todo(this.TodoId, this.Name, this.Description, this.Done);

  factory Todo.fromJson(Map<String, dynamic> json) {
      return Todo(
        json['todoId'],
        json['name'],
        json['description'],
        json['done']
      );
  }

    Map<String, dynamic> toJson() {
      return {
        'todoId': TodoId,
        'name': Name,
        'description': Description,
        'done': Done
      };
    }

    @override
    bool operator ==(Object other) {
      return other is Todo &&
        runtimeType == other.runtimeType &&
        TodoId == other.TodoId &&
        Name == other.Name &&
        Description == other.Description &&
        Done == other.Done;
    }
}